src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/CopyAatfControllerTests.cs

```diff
@@ -0,0 +1,311 @@
+﻿namespace EA.Weee.Web.Tests.Unit.Areas.Admin.Controllers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using System.Web.Mvc;
+    using AutoFixture;
+    using EA.Prsd.Core.Domain;
+    using EA.Prsd.Core.Mapper;
+    using EA.Weee.Api.Client;
+    using EA.Weee.Core.AatfReturn;
+    using EA.Weee.Core.Organisations;
+    using EA.Weee.Core.Shared;
+    using EA.Weee.Requests.AatfReturn;
+    using EA.Weee.Requests.Admin;
+    using EA.Weee.Security;
+    using EA.Weee.Web.Areas.Admin.Controllers;
+    using EA.Weee.Web.Areas.Admin.ViewModels.CopyAatf;
+    using EA.Weee.Web.Filters;
+    using EA.Weee.Web.Services;
+    using EA.Weee.Web.Services.Caching;
+    using FakeItEasy;
+    using FluentAssertions;
+    using Web.Infrastructure;
+    using Xunit;
+
+    public class CopyAatfControllerTests
+    {
+        private readonly Fixture fixture;
+        private readonly IMapper mapper;
+        private readonly IWeeeClient weeeClient;
+        private readonly IList<CountryData> countries;
+        private readonly BreadcrumbService breadcrumbService;
+        private readonly IWeeeCache cache;
+        private readonly CopyAatfController controller;
+
+        public CopyAatfControllerTests()
+        {
+            fixture = new Fixture();
+            mapper = A.Fake<IMapper>();
+            weeeClient = A.Fake<IWeeeClient>();
+            countries = A.Dummy<IList<CountryData>>();
+            breadcrumbService = A.Fake<BreadcrumbService>();
+            cache = A.Fake<IWeeeCache>();
+
+            controller = new CopyAatfController(() => weeeClient, breadcrumbService, mapper, cache);
+        }
+
+        [Fact]
+        public void ControllerMustHaveAuthorizeClaimsAttribute()
+        {
+            typeof(CopyAatfController).Should().BeDecoratedWith<AuthorizeInternalClaimsAttribute>(a => a.Match(new AuthorizeInternalClaimsAttribute(Claims.InternalAdmin)));
+        }
+
+        [Theory]
+        [InlineData(FacilityType.Aatf, "Copy AATF for new compliance year")]
+        public async Task CopyGet_CanEdit_SetsInternalBreadcrumb(FacilityType facilityType, string expectedBreadcrumb)
+        {
+            var aatf = fixture.Build<AatfData>().With(a => a.CanEdit, true).Create(); 
+            aatf.FacilityType = facilityType;
+
+            var aatfViewModel = fixture.Create<CopyAatfViewModel>();
+
+            A.CallTo(() => weeeClient.SendAsync(A<string>._, A<GetAatfById>._)).Returns(aatf);
+
+            A.CallTo(() => mapper.Map<CopyAatfViewModel>(aatf)).Returns(aatfViewModel);
+
+            await controller.CopyAatfDetails(aatf.Id);
+
+            Assert.Equal(expectedBreadcrumb, breadcrumbService.InternalActivity);
+            A.CallTo(() => weeeClient.SendAsync(A<string>._, A<GetAatfById>._)).MustHaveHappenedOnceExactly();
```

Comment:
Dont bother with this and the below check if this is just testing the breadcrumb